test/behavior/nesting.ts

```diff
@@ -416,6 +417,49 @@ async function shouldBehaveLikeNesting(
       );
     });
 
+    it('can recursively burn nested token with the right number of recursive burns', async function () {
+      // Parent
+      // -> Child1
+      //      -> GrandChild1
+      //      -> GrandChild2
+      //        -> GreatGrandChild1
+      // -> Child2
+      // Total tree 5 (4 recursive burns)
+      const childId = await nestMint(child, parent.address, parentId);
+      const childId2 = await nestMint(child, parent.address, parentId);
+      const grandChild1 = await nestMint(child, child.address, childId);
+      const grandChild2 = await nestMint(child, child.address, childId);
+      const greatGrandChild2 = await nestMint(child, child.address, grandChild2);
+      await parent.connect(tokenOwner).acceptChild(parentId, 0);
+      await parent.connect(tokenOwner).acceptChild(parentId, 0);
+      await child.connect(tokenOwner).acceptChild(childId, 0);
+      await child.connect(tokenOwner).acceptChild(childId, 0);
+      await child.connect(tokenOwner).acceptChild(grandChild2, 0);
+
+      // 0 is not enough
+      await expect(parent.connect(tokenOwner)['burn(uint256,uint256)'](parentId, 0))
+        .to.be.revertedWithCustomError(parent, 'RMRKMaxRecursiveBurnsReached')
+        .withArgs(child.address, childId);
+      // 1 is not enough
+      await expect(parent.connect(tokenOwner)['burn(uint256,uint256)'](parentId, 1))
+        .to.be.revertedWithCustomError(parent, 'RMRKMaxRecursiveBurnsReached')
+        .withArgs(child.address, grandChild1);
+      // 2 is not enough
+      await expect(parent.connect(tokenOwner)['burn(uint256,uint256)'](parentId, 2))
```

Comment:
It would look a bit nicer if typesafe calls were used here but it's not a big deal

Reply:
Yep, added task. This is something I'd like to. We need to figure out how to do it from behavior suites.